src/foundry/client/pixi/layers/effects/weather-effects.d.ts

```diff
@@ -0,0 +1,167 @@
+export {};
+
+declare global {
+  /**
+   * A CanvasLayer for displaying visual effects like weather, transitions, flashes, or more
+   */
+  class WeatherEffects extends FullCanvasObjectMixin(CanvasLayer) {
+    /**
+     * @defaultValue `true`
+     */
+    override sortableChildren: boolean;
+
+    /**
+     * @defaultValue `"none"`
+     */
+    override eventMode: PIXI.EventMode;
+
+    /**
+     * Sorting values to deal with ties.
+     * @defaultValue `1000`
+     */
+    static PRIMARY_SORT_ORDER: number;
+
+    /**
+     * @defaultValue `foundry.utils.mergeObject(super.layerOptions, {name: "effects"})`
+     */
+    static override get layerOptions(): CanvasLayer.LayerOptions;
+
+    /**
+     * Array of weather effects linked to this weather container.
+     */
+    effects: Map<string, (ParticleEffect | WeatherShaderEffect)[]>;
```

Comment:
Nit, I prefer `Map<string, Array<ParticleEffect | WeatherShaderEffect>>` with a union it's "clearer" to me but it's really minor and just my opinion.